wallet-api/src/main/java/org/exoplatform/wallet/model/reward/RewardPeriodType.java

```diff
@@ -35,9 +35,9 @@ public RewardPeriod getPeriodOfTime(LocalDateTime localDateTime) {
     switch (this) {
     case WEEK:
       LocalDateTime firstDayOfThisWeek = localDateTime.with(DayOfWeek.MONDAY);
-      LocalDateTime firstDayOfNextWeek = firstDayOfThisWeek.plusWeeks(1);
+      LocalDateTime lastDayOfThisWeek = firstDayOfThisWeek.plusDays(6);
```

Comment:
This is not the fix to use for reward computing. The problem comes from gamification which relies on Date and not timestamp. Anyway, to fix this, a wekk should be of 7 days and not 6, thus, the workaround about Gamification Modeling problem is simply this code :
https://github.com/Meeds-io/wallet/commit/d3837d8482aa5455ff4be2948e84cf6f49e3c584#diff-e1260f9635242f9987ad47e0ac27dadf65f4d982f1382314d281e976aeaee119R258

so please revert this, I will take the issue and make a dedicated PR for it. cc @AzmiTouil @GouadriaHanen  FYI.

Reply:
OK. Done!